diff -up xdrawchem-1.9.9/xdrawchem/bracket.cpp.warn xdrawchem-1.9.9/xdrawchem/bracket.cpp
--- xdrawchem-1.9.9/xdrawchem/bracket.cpp.warn	2005-06-09 20:23:57.000000000 +0200
+++ xdrawchem-1.9.9/xdrawchem/bracket.cpp	2010-11-25 01:15:32.843249035 +0100
@@ -253,7 +253,10 @@ Drawable * Bracket::FindNearestObject(DP
     dist = dist1;
   else
     dist = dist2;
-  delete tl, tr, bl, br;
+  delete tl;
+  delete tr;
+  delete bl;
+  delete br;
   return this;
 }
 
diff -up xdrawchem-1.9.9/xdrawchem/gobject.cpp.warn xdrawchem-1.9.9/xdrawchem/gobject.cpp
--- xdrawchem-1.9.9/xdrawchem/gobject.cpp.warn	2005-06-13 15:02:09.000000000 +0200
+++ xdrawchem-1.9.9/xdrawchem/gobject.cpp	2010-11-25 01:14:58.399000590 +0100
@@ -227,7 +227,10 @@ Drawable * GraphicObject::FindNearestObj
     dist = dist1;
   else
     dist = dist2;
-  delete tl, tr, bl, br;
+  delete tl;
+  delete tr;
+  delete bl;
+  delete br;
   return this;
 }
 
